packages/offix-cache/src/cache/mutations.ts

```diff
@@ -53,113 +96,112 @@ export const createMutationOptions = <T = {
  * Generate the update function to update the cache for a given operation and query.
- * Ignores the scenario where the cache operation is an update as this is handled automatically
- * from Apollo Client 2.5 onwards.
- * @param operation The title of the operation being performed
- * @param idField The id field the item keys off
- * @param updateQuery The Query to update in the cache
- * @param operationType The type of operation being performed. Defaults to "add".
- */
+
+ **/
 export const getUpdateFunction = (options: CacheUpdateOptions): MutationUpdaterFn => {
+  if (!options.updateQuery) {
+    throw new Error("Required updateQuery parameter is not supplied");
+  }
+  if (!options.mutationName) {
+    throw new Error("Required mutationName parameter is not supplied");
+  }
   const {
-    operation,
+    mutationName,
     updateQuery,
     operationType = CacheOperation.ADD,
     idField = "id"
   } = options;
-  if (!updateQuery) {
-    return () => {
-      return;
-    };
-  }
 
   const { query, variables } = deconstructQuery(updateQuery);
   const queryField = getOperationFieldName(query);
-
   let updateFunction: MutationUpdaterFn;
-
   switch (operationType) {
     case CacheOperation.ADD:
       updateFunction = (cache, { data }) => {
-        try {
-          if (data) {
-            let queryResult = cache.readQuery({ query, variables }) as any;
-            const operationData = data[operation];
-            const result = queryResult[queryField];
-            if (result && operationData) {
-              if (!result.find((item: any) => {
-                return item[idField] === operationData[idField];
-              })) {
-                result.push(operationData);
-              }
-            } else {
-              queryResult = [result];
+        let queryResult;
+        if (data) {
+          const operationData = data[mutationName];
+          try {
+            queryResult = cache.readQuery({ query, variables }) as any;
+          } catch (e) {
+            queryResult = {};
+          }
+
+          const result = queryResult[queryField];
+          if (result && operationData) {
+            // FIXME deduplication should happen on subscriptions
+            // We do that every time no matter if we have subscription
+            if (!result.find((item: any) => {
+              return item[idField] === operationData[idField];
+            })) {
+              result.push(operationData);
             }
-            cache.writeQuery({
-              query,
-              variables,
-              data: queryResult
-            });
+          } else {
+            queryResult[queryField] = [operationData];
           }
-        } catch (e) {
-          console.info(e);
+          cache.writeQuery({
+            query,
+            variables,
+            data: queryResult
+          });
         }
+
       };
       break;
     case CacheOperation.DELETE:
       updateFunction = (cache, { data }) => {
-        try {
```

Comment:
any reason you removed the try from here? any cache action should be wrapped in a try/catch block surely.

Reply:
ah.. right. I removed it because the only case I found for failure was on reading data, but then delete can also fail on read. This is bug